packages/flutter/lib/src/widgets/app.dart

```diff
@@ -1518,6 +1518,7 @@ class _WidgetsAppState extends State<WidgetsApp> with WidgetsBindingObserver {
     if (_usesRouter) {
       assert(_effectiveRouteInformationProvider != null);
       routing = Router<Object>(
+        restorationId: 'router',
```

Comment:
Any possibility for collision with user-defined restorationIds?  I assume we also have other hardcoded restorationIds in the framework like this elsewhere.

Reply:
This ID is in the restoration bucket scope of the WidgetApp and within it it is unique.